test/integration/offline_test.go

```diff
@@ -0,0 +1,62 @@
+// +build integration
+
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+)
+
+func TestOffline(t *testing.T) {
+	MaybeParallel(t)
+
+	t.Run("group", func(t *testing.T) {
+		for _, runtime := range []string{"docker", "crio", "containerd"} {
+			t.Run(runtime, func(t *testing.T) {
+				MaybeParallel(t)
+				WaitForStartSlot(t)
+
+				if runtime != "docker" && NoneDriver() {
+					t.Skipf("skipping %s - incompatible with none driver", t.Name())
+				}
+
+				profile := UniqueProfileName(fmt.Sprintf("offline-%s", runtime))
+				ctx, cancel := context.WithTimeout(context.Background(), 15*time.Minute)
+				defer CleanupWithLogs(t, profile, cancel)
+
+				startArgs := []string{"start", "-p", profile, "--alsologtostderr", "-v=1", "--wait=true", "--container-runtime", runtime}
+				startArgs = append(startArgs, StartArgs()...)
+				c := exec.CommandContext(ctx, Target(), startArgs...)
+				env := os.Environ()
+				env = append(env, "HTTP_PROXYS=172.1.1.1")
```

Comment:
was this supposed to be HTTP_PROXY ?

Reply:
This should be HTTPS, since that's the protocol which images are pulled by currently.

Reply:
Maybe the env name should be `HTTPS_PROXY`, rather than `HTTP_PROXYS`, is this a typo?

Reply:
These variables are not defined by minikube, but are rather a common standard used by multiple commands (wget, curl, go-lang). Here's how Go uses it:

https://github.com/golang/net/blob/c0dbc17a35534bf2e581d7a942408dc936316da4/http/httpproxy/proxy.go#L91

Reply:
Yes, I mean the env name  should be `HTTPS_PROXY`, and in this testcase, you wrote `HTTP_PROXYS=172.1.1.1`, should be `HTTPS_PROXY=172.1.1.1`?